src/WebClient.spec.js

```diff
@@ -201,18 +201,28 @@ describe('WebClient', function () {
 
         // intentially vague about the method name
         return this.client.apiCall('upload', {
-            file: imageBuffer,
-            filename: 'train.png',
-          })
+          file: imageBuffer,
+          filename: 'train.png',
+        })
+          .then((parts) => {
+            assert.lengthOf(parts.files, 1);
+            const file = parts.files[0];
+            // options were not provided to the form builder
+            assert.include(file, { fieldname: 'file' });
+          });
+      });
+
+      it('should properly serialize when the file is an object with value as a Buffer and options containing filename', function () {
+        const imageBuffer = fs.readFileSync(path.resolve('test', 'fixtures', 'train.jpg'));
+
+        // intentially vague about the method name
+        return this.client.apiCall('upload', {
+          file: { value: imageBuffer, options: { filename: 'train.png' } },
+          filename: 'train.png',
```

Comment:
i was trying to avoid this usage all together, because it seems redundant to have to specify the filename twice. can you see any advantage to doing it this way?

Reply:
The main advantage I see is the current implementation state of all the folks who used the old style. I don't know whether there is a need for someone to provide further attributes in the  options field (e.g. contentType or knownLength)

Reply:
i also want to make sure the implementation accounts for those who used the old style, but i prefer that the path we take to get there is not to redo the same hack, but to smooth out the API so that you don't need to redundantly specify the filename.

i wouldn't call it a hack if there was legitimately a use case or need for using the `{ value, options }` format for `file`. i don't see any benefit from exposing `contentType` and/or `knownLength` at this time, since that only has the potential to force to incorrect values and cause the upload to fail (but I'm happy to be corrected if there is a real use case).

maybe we can channel this effort towards investigating why the following code didn't behave as we want: https://github.com/slackapi/node-slack-sdk/blob/a206c3e1286be958c963795f9ce80a18cb85ce7d/src/WebClient.ts#L530-L537

AFAICT it should result in the same thing...

also, have you tried this implementation to upload actual files? i'm not 100% confident that the nock expectations are correct.

Reply:
Sorry, but I don't really understand your comment... this code snippet, you mentioned above works now as well. Off course I have tried it before I've opened this PR. And the problem with it, why it didn't work before was in the implementation of 
https://github.com/slackapi/node-slack-sdk/blob/a206c3e1286be958c963795f9ce80a18cb85ce7d/src/WebClient.ts#L556-L561

I've added this part in order to provide the attribute `options` to the form builder.
```
if (key === 'file' && value.value) {
    form.append(key, value.value, value.options);
    return form;
}
```
https://github.com/KharitonOff/node-slack-sdk/blob/45c77022a81cd2631a5a2bd3ee0dd29e401f51d7/src/WebClient.ts#L554-L563

This is actually the main goal of this PR and all other things we talked here about, were done with intention not to break the current state. As I mentioned in the PR description, you can upload files now with no need of additional/redundant information
```
let file = fs.readFileSync('../file.png')
slackClient.files.upload({ channels, file, filename: 'file.png' })
```

Should i remove the 'backup' solution to restrict the file upload to only one way?

Reply:
@KharitonOff thanks so much for your clarification. i missed that diff amidst the rest of the changes but i now see where the issue was before.

yes, i'd prefer removing the backup solution so that we don't unnecessarily expand the API contract. if you can do that, i'm happy to land this PR!